src/multi_vector_simulator/C0_data_processing.py

```diff
@@ -1529,6 +1529,12 @@ def receive_timeseries_from_csv(
     if input_type == "input":
         compute_timeseries_properties(dict_asset)
 
+    if TIMESERIES in dict_asset:
+        for i in range(0, len(dict_asset[TIMESERIES])):
+            if dict_asset[TIMESERIES][i] == None:
+                logging.Error("Nan value found. Changing Nan value to 0")
+                dict_asset[TIMESERIES][i] = 0  # TIMESERIES.filna(0)
```

Comment:
Analogeously with the assertions in your other PR and @mahendrarks, I would simplify to
```suggestion
    if TIMESERIES in dict_asset:
        if (dict_asset[TIMESERIES] == None).any():
                incidents = sum(dict_asset[TIMESERIES] == None)
                logging.Error(f"A number of {incidents} NaN value(s) found in the {TIMESERIES} of {dict_asset[LABEL]}. Changing NaN value(s) to 0.")
                dict_asset.update({TIMESERIES: dict_asset[TIMESERIES].filna(0)})
```
The other probably also works, but it needs more processing time (looping though manually always takes more time than using an inbuilt function).

I improved the logging message, so that it can be connected to a specific asset.

I also added `incidents` to try and count the number of incidents that there is a NaN. I think this should work, as `sum(boolean_list)` should be a `int`.